vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java

```diff
@@ -59,147 +154,343 @@ private void addFocusShortcut() {
         // begin-source-example
-        // source-example-heading: Text field with autoselect
-        TextField textField = new TextField();
-        textField.setLabel("Text field label");
-        textField.setValue("Text field value");
-        textField.setAutoselect(true);
+        // source-example-heading: Maximum height
+        TextArea textArea = new TextArea("Description");
+        textArea.getStyle().set("maxHeight", "150px");
+        textArea.setPlaceholder("Write here ...");
+        // end-source-example
 
+        textArea.getStyle().set("padding", "0");
+        textArea.setId("text-area-with-max-height");
+        addCard("Text Area", "Maximum height", textArea);
+    }
+
+    private void textAreaMinHeight() {
+        // begin-source-example
+        // source-example-heading: Minimum height
+        TextArea textArea = new TextArea("Description");
+        textArea.getStyle().set("minHeight", "150px");
+        textArea.setPlaceholder("Write here ...");
         // end-source-example
 
-        addCard("Text field with autoselect", textField);
+        textArea.getStyle().set("padding", "0");
+        textArea.setId("text-area-with-min-height");
+        addCard("Text Area", "Minimum height", textArea);
     }
 
-    private void addNumberFields() {
+    private void prefixAndSuffix() {
+        Div div = new Div();
         // begin-source-example
-        // source-example-heading: Number fields
+        // source-example-heading: Currency field
         NumberField dollarField = new NumberField("Dollars");
         dollarField.setPrefixComponent(new Span("$"));
 
         NumberField euroField = new NumberField("Euros");
-        euroField.setSuffixComponent(new Span("€"));
-
-        NumberField stepperField = new NumberField("Stepper");
-        stepperField.setValue(1d);
-        stepperField.setMin(0);
-        stepperField.setMax(10);
-        stepperField.setHasControls(true);
-
         euroField.setSuffixComponent(new Span("€"));
         // end-source-example
 
         dollarField.setId("dollar-field");
         euroField.setId("euro-field");
-        stepperField.setId("step-number-field");
+        div.add(dollarField, new Text(" "), euroField);
+        addCard("Prefix and suffix", "Currency field", div);
+    }
+
+    private void prefixAndSuffixSearch() {
+        // begin-source-example
+        // source-example-heading: Search field
+        TextField textField = new TextField();
+        textField.setPlaceholder("Search");
+        Icon icon = VaadinIcon.SEARCH.create();
+        textField.setPrefixComponent(icon);
+        // end-source-example
 
-        addCard("Number fields", dollarField, euroField, stepperField);
+        textField.setId("text-field-search-id");
+        icon.setId("icon-id");
+        addCard("Prefix and suffix", "Search field", textField);
     }
 
-    private void addEmailFieldFields() {
-        Div message = new Div();
+    private void validationRequired() {
```

Comment:
All the validation demos need to be reviewed after we have some consensus of how the issues should be fixed and what are the preferred ways to do validation.